src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/XplatEventLogger.cs

```diff
@@ -18,8 +18,13 @@ internal sealed partial class XplatEventLogger : EventListener
     {
         public XplatEventLogger() { }
 
-        private static readonly Lazy<string?> eventSourceNameFilter = new Lazy<string?>(() => CompatibilitySwitch.GetValueInternal("EventSourceFilter"));
-        private static readonly Lazy<string?> eventSourceEventFilter = new Lazy<string?>(() => CompatibilitySwitch.GetValueInternal("EventNameFilter"));
+        private static readonly Lazy<string?> eventSourceNameFilter = new Lazy<string?>(() => GetClrConfig("EventSourceFilter"));
```

Comment:
You can drop `Lazy` while you are on it. The allocation and initialization of `Lazy<T>` costs about as much as just fetching the config string.